tests/functional/cross_db_utils/test_cross_db.py

```diff
@@ -0,0 +1,72 @@
+import os
+import pytest
+from dbt.tests.util import run_dbt
+from tests.functional.cross_db_utils.fixtures import (
+    macros__test_assert_equal_sql,
+    seeds__data_dateadd_csv,
+    models__test_dateadd_sql,
+    models__test_dateadd_yml,
+    seeds__data_datediff_csv,
+    models__test_datediff_sql,
+    models__test_datediff_yml,
+)
+
+
+class BaseCrossDbMacro:
+    # install this repo as a package!
+    @pytest.fixture(scope="class")
+    def packages(self):
+        return {"packages": [{"local": os.getcwd()}]}
+
+    # setup
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"test_assert_equal.sql": macros__test_assert_equal_sql}
+    
+    # each child class will reimplement 'models' + 'seeds'
+    def seeds(self):
+        return {}
+        
+    def models(self):
+        return {}
+
+    # actual test sequence
+    def test_build_assert_equal(self, project):
+        run_dbt(['deps'])
+        run_dbt(['build'])    # seed, model, test -- all handled by dbt
```

Comment:
This is actually running dbt, and making sure the thing passes. We could also:
- add `assert` statements
- inspect results / logs
- ensure that dbt **fails** if given bad input, via `expect_pass=False`